internal/pkg/diagnostics/diagnostics.go

```diff
@@ -352,29 +353,58 @@ func writeRedacted(errOut, resultWriter io.Writer, fullFilePath string, fileResu
 	return err
 }
 
+const redactionMarkerPrefix = "__mark_redact_"
+
 // redactMap sensitive values from the underlying map
 // the whole generic function here is out of paranoia. Although extremely unlikely,
 // we have no way of guaranteeing we'll get a "normal" map[string]interface{},
 // since the diagnostic interface is a bit of a free-for-all
-func redactMap[K comparable](errOut io.Writer, inputMap map[K]interface{}) map[K]interface{} {
+func redactMap[K comparable](errOut io.Writer, inputMap map[K]interface{}, sliceElem bool) map[K]interface{} {
 	if inputMap == nil {
 		return nil
 	}
+
+	redactionMarkers := []string{}
 	for rootKey, rootValue := range inputMap {
 		if rootValue != nil {
 			switch cast := rootValue.(type) {
 			case map[string]interface{}:
-				rootValue = redactMap(errOut, cast)
+				rootValue = redactMap(errOut, cast, sliceElem)
 			case map[interface{}]interface{}:
-				rootValue = redactMap(errOut, cast)
+				rootValue = redactMap(errOut, cast, sliceElem)
 			case map[int]interface{}:
-				rootValue = redactMap(errOut, cast)
+				rootValue = redactMap(errOut, cast, sliceElem)
+			case []interface{}:
+				// Recursively process each element in the slice so that we also walk
+				// through lists (e.g. inputs[4].streams[0]). This is required to
+				// reach redaction markers that are inside slice items. Set SliceElem to true
+				// to avoid global redaction of slice elements.
+				for i, value := range cast {
+					switch m := value.(type) {
+					case map[string]interface{}:
+						cast[i] = redactMap(errOut, m, true)
+					case map[interface{}]interface{}:
+						cast[i] = redactMap(errOut, m, true)
+					case map[int]interface{}:
+						cast[i] = redactMap(errOut, m, true)
```

Comment:
Nit: you could collapse these three cases into one since they're doing the same thing.

```suggestion
					case map[string]interface{}, map[interface{}]interface{}, map[int]interface{}:
						cast[i] = redactMap(errOut, m, true)
```